ports/stm32/Makefile

```diff
@@ -103,6 +103,9 @@ MICROPY_FLOAT_IMPL = double
 endif
 endif
 
+# Configure mboot if required
+include mboot/mbootenc.mk
```

Comment:
What does "enc" mean?

I don't think it's necessary to have another .mk file.  I think it's best to just have the VID/PID set by the usual board's `mpconfigboard.mk` file, so the settings for a board are all in one place.

Then we just need to set a default VID/PID in this `Makefile`.  But actually that's already done with the `DEVICE` variable.  I suggest this variable be replaced with `USB_DEVICE_VID` and `USB_DEVICE_PID` and they can be overridden by a boards .mk file.

Reply:
Ah, I first made that makefile as part of the encrypted work, sort of forgot I'd named it as such.

With the encryption work, there were a number of settings I wanted to share between mboot and the main firmware and thought it might be better to have them in one file rather than duplicating them in both makefiles, particularly the length of the firmware footer struct:
https://github.com/micropython/micropython/pull/5309/files#diff-7b014da95293e2883b0b4f53598b5cbf

Seeing as I'd already added this makefile chunk I just used it for the vid/pid as well, but yes it seems a bit pointless for just this one pair of defines.